### PR TITLE
HPCC-16858 Detect compressed files when reattaching XREF files.

### PR DESCRIPTION
**Pull request by jakesmith**
Check physical files, test to see if compressed so as to mark
the meta data correctly, also add size or compressedSize
dependent on whether compressed or not.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

### CODE REVIEW

**Comment by jakesmith** [comment]
@AttilaVamos - please review

File: system/jlib/jlzw.cpp [hunk 1]
```diff
@@ -2326,37 +2326,63 @@ class CCompressedFile : implements ICompressedFileIO, public CInterface
...
+
+ICompressedFileIO *createCompressedFileReader(IFileIO *fileio,IExpander *expander)
+{
+    CompressedFileTrailer trailer;
+    if (isCompressedFile(fileio, &trailer))
+    {
+        if (expander&&(trailer.recordSize!=0)) {
+            throw MakeStringException(-1, "Compressed file format error(%d), Encrypted?",trailer.recordSize);
+        }
+        unsigned compMethod1 = COMPRESS_METHOD_LZW;
```
Comment (AttilaVamos) [inline system/jlib/jlzw.cpp]: Is there any meaning of numbering compMethod variable?

**Review by AttilaVamos — APPROVED** [review]
Looks good. I have only one question.

**Comment by AttilaVamos** [comment]
@jakesmith back to you.

File: system/jlib/jlzw.cpp (hunk 1 above)
Comment (jakesmith) [inline system/jlib/jlzw.cpp]: No particular reason this temp variable is called 'compMethod*1*', no.
It was moved code, I don't know why it was called that.
I'll rename it.

**Comment by jakesmith** [comment]
@AttilaVamos - renamed compMethod1 to compMethod.

Please check , then I'll squash.

**Comment by AttilaVamos** [comment]
@jakesmith it seems good to merge.

**Comment by jakesmith** [comment]
Squashed.

@richardkchapman - ready to merge.

File: system/jlib/jlzw.cpp [hunk 2]
```diff
@@ -2379,12 +2405,12 @@ ICompressedFileIO *createCompressedFileReader(IFile *file,IExpander *expander, b
                         if (expander&&(trailer.recordSize!=0)) {
```
Comment (richardkchapman) [inline system/jlib/jlzw.cpp]: Feels like this code and that in the method above should be common?

**Comment by richardkchapman** [comment]
@jakesmith One comment

**Comment by jakesmith** [comment]
@richardkchapman - have commoned up some of the comp. type code. Please check and then I'll squash.

**Comment by richardkchapman** [comment]
@jakesmith Looks better

**Comment by jakesmith** [comment]
@richardkchapman - don't merge, I've spotted a bug.

**Comment by jakesmith** [comment]
@richardkchapman - fixed in above commit, also ensure logical file size was set (which was spotted by @cloLN in HPCC-16938).

Please review and I'll squash commits.

File: dali/dfuXRefLib/XRefFilesNode.cpp [hunk 3]
```diff
@@ -360,8 +364,39 @@ bool CXRefFilesNode::AttachPhysical(const char *Partmask,IUserDescriptor* udesc,
...
-        fileDesc->setPart(part.getPropInt("Num")-1,node,_filename.str());
+        unsigned partNo = part.getPropInt("Num")-1;
+
+        RemoteFilename rfn;
+        rfn.setPath(node->endpoint(), remoteFilePath);
+        Owned<IFile> iFile = createIFile(rfn);
+        offset_t physicalSize = iFile->size();
+        bool partCompressed = isCompressedFile(iFile);
+        if (first)
+            isCompressed = partCompressed;
```
Comment (mckellyln) [inline dali/dfuXRefLib/XRefFilesNode.cpp]: Should first be set to false here ?

Comment (jakesmith) [inline dali/dfuXRefLib/XRefFilesNode.cpp]: Yes, it should, will correct.

**Comment by jakesmith** [comment]
@mckellyln - added missing first = false, squashed commits and rebsaed on to latest candidate-6.4.0

Please check.

**Comment by mckellyln** [comment]
@jakesmith looks fine.

**Comment by jakesmith** [comment]
Thanks.
@richardkchapman - please merge.

**Pull request merged**